CMakeLists.txt

```diff
@@ -532,6 +530,57 @@ while(library_variants_copy)
             ${INSTALL_DIR}/lib/${target} ${install_directory}/lib
     )
 
+    #picolibc tests
+    string(REPLACE " " ";" clang_compile_flags ${runtimes_flags})
+    add_custom_target(check-picolibc_${variant})
+    set(picolibc_test_ld ${picolibc_SOURCE_DIR}/scripts/test-arm.ld)
+    if(${variant} EQUAL "aarch64")
+        set(picolibc_test_ld ${picolibc_SOURCE_DIR}/scripts/test-aarch64.ld)
+    endif()
+    set(picolibc_tests printf_scanf
+                        printf-tests
+                        rand
+                        regex
+                        ungetc
+                        fenv
+                        long_double
+                        malloc
+                        ffs
+                        setjmp
+                        atexit
+                        on_exit
+                        math-funcs
+                        timegm
+                        time-tests
+                        test-strtod
+                        test-strchr
+                        test-memset
+                        test-put
+                        test-efcvt
+                        malloc_stress
+                        posix-io)
+    foreach(test IN LISTS picolibc_tests)
+        add_custom_target(
+            build-check-picolibc_${variant}_${test}
+            COMMAND ${llvm_bin}/clang --config ${llvm_bin}/${variant}_semihost.cfg
+                -T ${install_directory}/lib/picolibc.ld -T ${picolibc_test_ld}
+                -nostdlib ${install_directory}/lib/libc.a ${install_directory}/lib/libg.a
+                ${install_directory}/lib/libm.a ${install_directory}/lib/libclang_rt.builtins.a
+                ${install_directory}/lib/libsemihost.a
+                ${picolibc_SOURCE_DIR}/test/${test}.c -o picolibc_test_${variant}_${test}.out
+            COMMENT "Building: ${test} for ${variant}"
+        )
+        add_custom_target(
+                check-picolibc_${variant}_${test}
+                ../samples/picolibc_test_wrapper.sh ${picolibc_test_command} picolibc_test_${variant}_${test}.out
+                DEPENDS picolibc_${variant} build-check-picolibc_${variant}_${test}
+                USES_TERMINAL
```

Comment:
`USES_TERMINAL` prevents tests running in parallel. The CMake docs say "With the [Ninja](https://cmake.org/cmake/help/latest/generator/Ninja.html#generator:Ninja) generator, this places the command in the console [pool](https://cmake.org/cmake/help/latest/prop_gbl/JOB_POOLS.html#prop_gbl:JOB_POOLS)" - I infer that only one job in the console pool can run at a time.